homeassistant/components/synology_dsm/__init__.py

```diff
@@ -164,7 +164,7 @@ async def async_coordinator_update_data_cameras() -> dict[
         surveillance_station = api.surveillance_station
 
         try:
-            async with async_timeout.timeout(10):
+            async with async_timeout.timeout(30):
```

Comment:
This is very long and will also delay startup of Home Assistant if the device is offline (because it is blocking during setup). Can you detect if it’s an NVR and if so, make it 30s ? Also what can take up to 30 seconds to reply, that sounds like some other issue

Reply:
the main problem is, that the [`surveillance_station.update`](https://github.com/mib1185/py-synologydsm-api/blob/ce6304bf779e2612bf523b5dc3be33d69f9e1f5e/src/synology_dsm/api/surveillance_station/__init__.py#L23) in the library does multiple API calls sequentially - first gather all cameras, next iterate over each camera and gather the motion detection state.
In case of #58793 the "gather all cameras" lasts already 2s which returns 7 cameras ... each "gather motion detection state" lasts ~1-1.5s, those the 7th iteration was already causing the timeout - honestly the used NAS (model DS214) is not quiet powerful, but also actual models rely often on efficient small ARM cpu's.

Maybe we can limit the timeout to 20s to be more friendly to the entrance class NAS - sure, the overall solution should be to divide the data gathering into own update coordinators per each camera, but this needs also a rework of the library (already on my [todo list](https://github.com/users/mib1185/projects/2/views/1) for the upcoming winter months)

Reply:
btw if the NAS is not online during setup a `ConfigEntryNotReady` is raised
https://github.com/home-assistant/core/blob/80f1e8770aac986e2f231e39447f1da7312ed00f/homeassistant/components/synology_dsm/__init__.py#L135

Reply:
I'm fine with accepting this now to relief some pressure for this release, but this is not the right fix. Instead of a global timeout it should have a timeout per request.